proposals/0000-swiftpm-binary-dependencies.md

```diff
@@ -173,21 +167,63 @@ We propose to support local and remote artifacts from the beginning. In the alte
+### PackageDescription
+To opt out of binary packages we propose a new configuration point inside the package description.
+
+```swift
+public final class Package {
+    ...
+    /// This disallows any binary dependency or any transitive binary dependency.
+    public var disallowsBinaryDependencies: Bool
+    ...
+}
+```
+
+## New `Package.resolved` Behavior
+
+TODO: This still needs to be investigated.
+
+### Resolution
+
+Package resolution and dependency expression will not be impacted by this change (except where explicitly noted).
+
+#### Multiple references to same artifact
+During resolution SwiftPM will check that all references to an artifact in a dependency graph have the same checksum.
+
+## Binary Target Artifact Format
+
+SwiftPM currently supports multiple platforms; however, this proposal only proposes to add support for binary targets on Apple platforms. The reason for this is that Apple platform provide ABI guarantees and an already existing format we can leverage to make simplify the initial implementation. For Apple platforms we propose to use the `XCFramework` format for artifacts. This format already supports dynamic and static linking. Furthermore, it can contain products for every individual Apple platform at once.
 
-**Note:** Should we also consider the core-libs-foundation ABI here?
+## Security
+
+When adding new external dependencies, it is always important to consider the security implication that it will bring with it. Comparing the trust level of a source-based to a binary-based dependency the first thought is that the trust level of the source-based dependency is higher since on can inspect its source code. However, there is no difference between a binary and source dependency since source-based dependencies can have security issues as well. One should have better reasons to trust a dependency than source being inspectable.
+
+There is still a significant difference between having a dependency with zero vs. any binary dependency. For example, the portability of a library with binary dependencies is far worse than the one with only source-based dependencies. For this reason, we propose to add an additional configuration point in the manifest that allows package authors to opt-out of binary dependencies.
+
+However, there are still some security related aspects when it comes to binary artifacts that we should mitigate. For example, when declaring a `binaryTarget` the hash of the artifact is required similar to Homebrew. By doing this an attacker needs to compromise both the server which provides the artifact as well as the git repository which provides the package manifest. A secondary reason is that the server providing the binary might be out of the package author's control and this way we can ensure that the expected binary is used.
+
+Lastly, the hash of the binary is stored in the package resolved to avoid that the vendor changes the artifact behind a version without anyone noticing.
```

Comment:
It may be preferable to define a whitelist configuration at the system level. These could include hosts and also paths so that a company could put in their hosts and when referencing public hosts like GitHub they can be limited to paths for specific orgs or repos. Often companies will clone a public repo onto their own servers and use those instead of the public version which could disappear or could be compromised in some way. Being able to import new releases, run through a security review, and publish internally is a much safer way to consume Open Source. And a whitelist could help ensure code bases are kept secure.